src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,24 +1611,1318 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+            return true;
+
+        if (Creature* agatha = player->FindNearestCreature(NPC_AGATHA_FENRIS, 10.0f, true))
+        {
+            if (agatha->GetOwner() == player)
+            {
+                if (Creature* fenrisStalker = player->FindNearestCreature(NPC_FENRIS_KEEP_STALKER, 50.0f, true))
+                {
+                    if (fenrisStalker->IsAIEnabled())
+                        fenrisStalker->AI()->SetGUID(player->GetGUID());
+                }
+            }
+        }
+
+        return true;
+    }
+};
+
+enum SummonFenrisActors
+{
+    SPELL_SUMMON_BLOODFANG_FENRIS               = 84054,
+    SPELL_SUMMON_CROWLEY_FENRIS                 = 84055,
+    SPELL_SUMMON_PHIN_ODELIC                    = 84056,
+    SPELL_SUMMON_BARTOLO_GINSETTI               = 84057,
+    SPELL_SUMMON_LOREMASTER_DIBBS               = 84058,
+    SPELL_SUMMON_MAGISTRATE_HENRY               = 84059,
+    SPELL_SUMMON_CARETAKER_SMITHERS             = 84060,
+    SPELL_SUMMON_SOPHIA                         = 84061
+};
+
+// 84053 - Summon Fenris Actors
+class spell_silverpine_summon_fenris_keep_actors : public SpellScript
+{
+    PrepareSpellScript(spell_silverpine_summon_fenris_keep_actors);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_SUMMON_BLOODFANG_FENRIS,
+            SPELL_SUMMON_CROWLEY_FENRIS,
+            SPELL_SUMMON_PHIN_ODELIC,
+            SPELL_SUMMON_BARTOLO_GINSETTI,
+            SPELL_SUMMON_LOREMASTER_DIBBS,
+            SPELL_SUMMON_MAGISTRATE_HENRY,
+            SPELL_SUMMON_CARETAKER_SMITHERS,
+            SPELL_SUMMON_SOPHIA
+        });
+    }
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        Unit* target = GetHitUnit();
+
+        target->CastSpell(target, SPELL_SUMMON_BLOODFANG_FENRIS, true);
+        target->CastSpell(target, SPELL_SUMMON_CROWLEY_FENRIS, true);
+        target->CastSpell(target, SPELL_SUMMON_PHIN_ODELIC, true);
+        target->CastSpell(target, SPELL_SUMMON_BARTOLO_GINSETTI, true);
+        target->CastSpell(target, SPELL_SUMMON_LOREMASTER_DIBBS, true);
+        target->CastSpell(target, SPELL_SUMMON_MAGISTRATE_HENRY, true);
+        target->CastSpell(target, SPELL_SUMMON_CARETAKER_SMITHERS, true);
+        target->CastSpell(target, SPELL_SUMMON_SOPHIA, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_silverpine_summon_fenris_keep_actors::HandleScriptEffect, EFFECT_0, SPELL_EFFECT_SCRIPT_EFFECT);
+    }
+};
+
+// 84065 - Despawn All Summons
+class spell_silverpine_despawn_all_summons_fenris_keep : public SpellScript
+{
+    PrepareSpellScript(spell_silverpine_despawn_all_summons_fenris_keep);
+
+    void HandleScriptEffect(SpellEffIndex /*effIndex*/)
+    {
+        Creature* target = GetHitCreature();
+
+        if (target->GetOwner() == GetCaster())
```

Comment:
check for caster being null, otherwise you'd remove all creatures not having an owner

Reply:
Removed spellscript as it's already implemented in an older PR.